File: app/src/main/java/uk/co/ourfriendirony/medianotifier/activities/ActivityFind.java
package uk.co.ourfriendirony.medianotifier.activities;

import android.os.AsyncTask;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.ListView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import uk.co.ourfriendirony.medianotifier.R;
import uk.co.ourfriendirony.medianotifier.activities.async.AddMediaItem;
import uk.co.ourfriendirony.medianotifier.activities.async.FindMediaItem;
import uk.co.ourfriendirony.medianotifier.clients.Client;
import uk.co.ourfriendirony.medianotifier.clients.ClientFactory;
import uk.co.ourfriendirony.medianotifier.db.Database;
import uk.co.ourfriendirony.medianotifier.db.DatabaseFactory;
import uk.co.ourfriendirony.medianotifier.db.PropertyHelper;

import static uk.co.ourfriendirony.medianotifier.general.Constants.INTENT_KEY;

public class ActivityFind extends AppCompatActivity {
    private EditText input;
    private ProgressBar progressBar;
    private ListView listView;
    private Client client;
    private Database db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        String intentKey = getIntent().getExtras().getString(INTENT_KEY);

        setTheme(PropertyHelper.getTheme(getBaseContext()));
        setContentView(R.layout.activity_find);
        getSupportActionBar().setTitle("Find " + intentKey);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        db = new DatabaseFactory().getDatabase(getApplicationContext(), intentKey);
        client = new ClientFactory().getClient(intentKey);

        input = (EditText) findViewById(R.id.find_input);
        progressBar = (ProgressBar) findViewById(R.id.find_progress);
        listView = (ListView) findViewById(R.id.find_list);

        input.setOnEditorActionListener(new TextView.OnEditorActionListener() {
            @Override
            public boolean onEditorAction(TextView textView, int actionId, KeyEvent event) {
                if (actionId == EditorInfo.IME_ACTION_SEND) {
                    String input = textView.getText().toString();
                    if (!"".equals(input)) {
                        new FindMediaItem(getBaseContext(), progressBar, listView, db, client).executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, input);
                    }
                    return true;
                } else {
                    return false;
                }
            }
        });

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                TextView textViewID = (TextView) view.findViewById(R.id.list_item_generic_id);
                TextView textViewTitle = (TextView) view.findViewById(R.id.list_item_generic_title);
                new AddMediaItem(getApplicationContext(), progressBar, db, client).execute(
                        textViewID.getText().toString(),
                        textViewTitle.getText().toString()
                );
            }
        });
    }
}

File: app/src/main/kotlin/uk/co/ourfriendirony/medianotifier/activities/ActivityFind.kt
package uk.co.ourfriendirony.medianotifier.activities

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.KeyEvent
import android.view.View
import android.view.inputmethod.EditorInfo
import android.widget.*
import android.widget.AdapterView.OnItemClickListener
import androidx.appcompat.app.AppCompatActivity
import uk.co.ourfriendirony.medianotifier.R
import uk.co.ourfriendirony.medianotifier.activities.async.AddMediaItem
import uk.co.ourfriendirony.medianotifier.activities.async.FindMediaItem
import uk.co.ourfriendirony.medianotifier.clients.Client
import uk.co.ourfriendirony.medianotifier.clients.ClientFactory
import uk.co.ourfriendirony.medianotifier.db.Database
import uk.co.ourfriendirony.medianotifier.db.DatabaseFactory
import uk.co.ourfriendirony.medianotifier.general.Constants.INTENT_KEY
import java.util.concurrent.Executors

class ActivityFind : AppCompatActivity() {
    private var input: EditText? = null
    private var progressBar: ProgressBar? = null
    private var listView: ListView? = null
    private var client: Client? = null
    private var db: Database? = null

    private val myHandler = Handler(Looper.getMainLooper())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_find)


        val intentKey = intent.extras!!.getString(INTENT_KEY)
        supportActionBar!!.title = "Find $intentKey"
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        db = DatabaseFactory().getDatabase(applicationContext, intentKey!!)
        client = ClientFactory().getClient(intentKey)
        input = findViewById(R.id.find_input)
        progressBar = findViewById(R.id.find_progress)
        listView = findViewById(R.id.find_list)

        input?.setOnEditorActionListener { textView: TextView, actionId: Int, _: KeyEvent? ->
            if (actionId == EditorInfo.IME_ACTION_SEND) {
                val input = textView.text.toString()
                if ("" != input) {
                    Executors.newSingleThreadExecutor().execute(
                        FindMediaItem(baseContext, progressBar, listView, db, client, myHandler, input)
                    )
                }
                true
            } else {
                false
            }
        }

        listView?.onItemClickListener = OnItemClickListener { _: AdapterView<*>?, view: View, _: Int, _: Long ->
            val id = view.findViewById<TextView>(R.id.list_item_generic_id).text.toString()
            val title = view.findViewById<TextView>(R.id.list_item_generic_title).text.toString()

            Executors.newSingleThreadExecutor().execute(
                AddMediaItem(applicationContext, progressBar, db, client, myHandler, id, title)
            )
        }
    }
}